app/models/one_login/ruby_saml/saml_generator.rb

```diff
@@ -128,9 +128,9 @@ def set_attribute_values(attr_name, hbx_enrollment)
         when 'Assigned QHP Identifier'
           hbx_enrollment.product.hios_id.gsub('-', '')
         when 'Total Amount Owed'
-          hbx_enrollment.total_premium - hbx_enrollment.applied_aptc_amount.to_f
+          (hbx_enrollment.total_premium.to_f - hbx_enrollment.applied_aptc_amount.to_f).round(2)
```

Comment:
This is purely an issue with instances of Float. We have been using FloatHelper's float_fix method to fix this. Can you use the same here instead of rounding the value by including the FloatHelper in the class?

```suggestion
          float_fix(hbx_enrollment.total_premium - hbx_enrollment.applied_aptc_amount.to_f)
```

Reply:
853c93b5e7a82db5ec41227a6711fe9fb0046792

Reply:
@saimekala07 We should be using `float_fix` and not `round_down_float_two_decimals` as this is Float error. The latter rounds the decimal down up to 2 decimals. In the example case from ticket, we want the paymentAmount to be 454.81 and not 454.80999999999995

Reply:
Updated!